src/plugins/intel_cpu/src/nodes/executors/reduce_list.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "executor.hpp"
+
+#include "reduce.hpp"
+#if defined(OV_CPU_WITH_ACL)
+#include "acl/acl_reduce.hpp"
+#endif
+#include "common/ref_reduce.hpp"
+
+#include "onednn/iml_type_mapper.h"
+#include "common/primitive_cache.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+struct ReduceExecutorDesc {
+    ExecutorType executorType;
+    ReduceExecutorBuilderCPtr builder;
+};
+
+const std::vector<ReduceExecutorDesc>& getReduceExecutorsList();
+
+class ReduceExecutorFactory : public ExecutorFactory {
+public:
+    ReduceExecutorFactory(const ReduceAttrs& reduceAttrs,
+                       const std::vector<MemoryDescPtr>& srcDescs,
+                       const std::vector<MemoryDescPtr>& dstDescs,
+                       const ExecutorContext::CPtr context) : ExecutorFactory(context) {
+        for (auto& desc : getReduceExecutorsList()) {
+            if (desc.builder->isSupported(reduceAttrs, srcDescs, dstDescs)) {
+                supportedDescs.push_back(desc);
+            }
+        }
+    }
+
+    ~ReduceExecutorFactory() = default;
+    virtual ReduceExecutorPtr makeExecutor(const ReduceAttrs& reduceAttrs,
+                                        const std::vector<MemoryDescPtr>& srcDescs,
+                                        const std::vector<MemoryDescPtr>& dstDescs,
+                                        const dnnl::primitive_attr &attr) {
+        auto build = [&](const ReduceExecutorDesc* desc) {
+            switch (desc->executorType) {
+#if defined(OPENVINO_ARCH_X86_64)
+                case ExecutorType::x64: {
+                    auto builder = [&](const JitReduceExecutor::Key& key) -> ReduceExecutorPtr {
+                        auto executor = desc->builder->makeExecutor(context);
+                        if (executor->init(reduceAttrs, srcDescs, dstDescs, attr)) {
+                            return executor;
+                        } else {
+                            return nullptr;
+                        }
+                    };
+
+                    auto key = JitReduceExecutor::Key(reduceAttrs, srcDescs, dstDescs, attr);
+                    auto res = runtimeCache->getOrCreate(key, builder);
+                    return res.first;
+                } break;
+#endif
```

Comment:
That part is not implemented yet since x64 executor is not moved. Lets just delete

Reply:
Removed.